components/tag-list/tag-list.js

```diff
@@ -39,41 +64,159 @@ class TagList extends ArrowKeysMixin(LitElement) {
 		/** @ignore */
 		this.arrowKeysDirection = 'leftrightupdown';
 		this._items = [];
+		this._resizeObserver = null;
+		this._showHiddenTags = false;
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		if (this._resizeObserver) this._resizeObserver.disconnect();
 	}
 
 	firstUpdated(changedProperties) {
 		super.firstUpdated(changedProperties);
 
-		requestAnimationFrame(() => {
-			this._items = this._getTagListItems();
-			this._items.forEach((item, index) => {
-				item.setAttribute('tabIndex', index === 0 ? 0 : -1);
-			});
-		});
+		this._container = this.shadowRoot.querySelector('.tag-list-outer-container');
+		this._resizeObserver = new ResizeObserver((e) => requestAnimationFrame(() => this._handleResize(e)));
+		this._resizeObserver.observe(this._container);
 	}
 
 	render() {
+		let hiddenCount = 0;
+		let hasHiddenTags = false;
+		this._items.forEach((element, index) => {
+			if (index >= this._chompIndex) hasHiddenTags = true;
+			if (!this._showHiddenTags && index >= this._chompIndex) {
+				hiddenCount++;
+				element.setAttribute('data-is-chomped', '');
+			} else {
+				element.removeAttribute('data-is-chomped');
+			}
+		});
+
+		let button = null;
+		if (hasHiddenTags) {
+			button = this._showHiddenTags ? html`
+				<d2l-button-subtle
+					class="d2l-tag-list-button"
+					@click="${this._toggleHiddenTagVisibility}"
+					@keydown="${this._handleButtonKeydown}"
```

Comment:
I think `@click` should be handling the ENTER case already just as part of a normal `<button>`'s API? Ideally folks (ourselves included) don't need special handlers for keyboard events on buttons.

I see we're also moving focus as part of the keyboard handling... hopefully we can either also do that as part of the `click` handling?

Reply:
Yup `@click` does handle enter. I'd thought there wasn't a way to check if it was a keydown vs click event from the event so I separated them but I was definitely wrong🤦‍♀️- will change.

We wouldn't want to move focus on click, correct? It seemed to make sense to me to do so on enter since otherwise the keyboard focus goes into a bit of a no-mans-land and this way it becomes easy to toggle.

Reply:
I think yeah ideally there isn't a visible focus change, but if we can move focus and the browser interprets it as part of the click and therefore triggers the invisible `:focus-visible` stuff, then that would be sweet.

Reply:
But also I think we do visibly move focus for similar scenarios when the user clicks -- e.g. closing a dialog and moving focus back to the opener. It's not quite the same but if it involved specialized code and wasn't too ugly it might be fine.